File: swift/Sources/Kiahk/CopticMonthsData.swift
import Foundation

/// One entry of the Coptic month-name table
/// (mirror of one entry in core/coptic_months.json).
public struct CopticMonthRecord: Equatable, Hashable, Sendable {
    public let month: Int
    public let names: [String: String]

    public init(month: Int, names: [String: String]) {
        self.month = month
        self.names = names
    }
}

/// Hand-maintained mirror of core/coptic_months.json.
/// Keep order identical (months 1..13) for cross-port test parity.
public let kCopticMonths: [CopticMonthRecord] = [
    CopticMonthRecord(month: 1,  names: ["en": "Thout",    "ar": "توت"]),
    CopticMonthRecord(month: 2,  names: ["en": "Paopi",    "ar": "بابة"]),
    CopticMonthRecord(month: 3,  names: ["en": "Hathor",   "ar": "هاتور"]),
    CopticMonthRecord(month: 4,  names: ["en": "Koiak",    "ar": "كيهك"]),
    CopticMonthRecord(month: 5,  names: ["en": "Tobi",     "ar": "طوبة"]),
    CopticMonthRecord(month: 6,  names: ["en": "Meshir",   "ar": "أمشير"]),
    CopticMonthRecord(month: 7,  names: ["en": "Paremhat", "ar": "برمهات"]),
    CopticMonthRecord(month: 8,  names: ["en": "Parmouti", "ar": "برمودة"]),
    CopticMonthRecord(month: 9,  names: ["en": "Pashons",  "ar": "بشنس"]),
    CopticMonthRecord(month: 10, names: ["en": "Paoni",    "ar": "بؤونة"]),
    CopticMonthRecord(month: 11, names: ["en": "Epip",     "ar": "أبيب"]),
    CopticMonthRecord(month: 12, names: ["en": "Mesori",   "ar": "مسرى"]),
    CopticMonthRecord(month: 13, names: ["en": "Nasie",    "ar": "نسيء"]),
]

File: csharp/Kiahk/CopticMonthsData.cs
using System.Collections.Generic;

namespace Kiahk;

/// <summary>One entry of the Coptic month-name table
/// (mirror of one entry in core/coptic_months.json).</summary>
public sealed record CopticMonthRecord(
    int Month,
    IReadOnlyDictionary<string, string> Names);

/// <summary>Hand-maintained mirror of core/coptic_months.json.
/// Keep order identical (months 1..13) for cross-port test parity.</summary>
public static class CopticMonthsData
{
    /// <summary>The 13 Coptic month names, in order 1..13.</summary>
    public static readonly IReadOnlyList<CopticMonthRecord> Months = new CopticMonthRecord[]
    {
        new(1,  new Dictionary<string, string> { ["en"] = "Thout",    ["ar"] = "توت" }),
        new(2,  new Dictionary<string, string> { ["en"] = "Paopi",    ["ar"] = "بابة" }),
        new(3,  new Dictionary<string, string> { ["en"] = "Hathor",   ["ar"] = "هاتور" }),
        new(4,  new Dictionary<string, string> { ["en"] = "Koiak",    ["ar"] = "كيهك" }),
        new(5,  new Dictionary<string, string> { ["en"] = "Tobi",     ["ar"] = "طوبة" }),
        new(6,  new Dictionary<string, string> { ["en"] = "Meshir",   ["ar"] = "أمشير" }),
        new(7,  new Dictionary<string, string> { ["en"] = "Paremhat", ["ar"] = "برمهات" }),
        new(8,  new Dictionary<string, string> { ["en"] = "Parmouti", ["ar"] = "برمودة" }),
        new(9,  new Dictionary<string, string> { ["en"] = "Pashons",  ["ar"] = "بشنس" }),
        new(10, new Dictionary<string, string> { ["en"] = "Paoni",    ["ar"] = "بؤونة" }),
        new(11, new Dictionary<string, string> { ["en"] = "Epip",     ["ar"] = "أبيب" }),
        new(12, new Dictionary<string, string> { ["en"] = "Mesori",   ["ar"] = "مسرى" }),
        new(13, new Dictionary<string, string> { ["en"] = "Nasie",    ["ar"] = "نسيء" }),
    };
}
